src/main/java/frc/robot/commands/testing/TestSubsystem.java

```diff
@@ -0,0 +1,171 @@
+package frc.robot.commands.testing;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.Subsystem;
+import frc.robot.devices.ColorSensor;
+import frc.robot.devices.Lemonlight;
+import frc.robot.devices.Lidar;
+import frc.robot.oi.drivers.ShuffleboardDriver;
+import frc.robot.utilities.Testable;
+import frc.robot.utilities.lists.Colors;
+import java.util.HashMap;
+
+/**
+ * Tests a testable subsystem.
+ */
+public class TestSubsystem extends CommandBase {
+
+    private CANSparkMax[] motors;
+    private ColorSensor[] colorSensors;
+    private Lemonlight[] limelights;
+    private Lidar[] lidars;
+    private RelativeEncoder[] encoders;
+    private double[] startingPositions;
+    private HashMap<String, Boolean> testStates;
+    private Testable toTest;
+    
+    private Timer timer;
+    
+    /**
+     * Tests a testable subsystem.
+     *
+     * @param toTest The testable object
+     */
+    public TestSubsystem(Testable toTest) {
+        this.toTest = toTest;
+
+        motors = toTest.getMotors();
+        colorSensors = toTest.getColorSensors();
+        limelights = toTest.getLimelights();
+        lidars = toTest.getLidars();
+
+        for (int i = 0; i < motors.length; i++) {
+            testStates.put("Motor " + motors[i].getDeviceId(), false);
+            encoders[i] = motors[i].getEncoder();
+        }
+
+        for (int i = 0; i < colorSensors.length; i++) {
+            testStates.put("Color Sensor " + i, false);
+        }
+
+        for (int i = 0; i < lidars.length; i++) {
+            testStates.put("Lidar " + i, false);
+        }
+
+        for (int i = 0; i < limelights.length; i++) {
+            testStates.put("Limelight " + i, false);
+        }
+
+        timer = new Timer();
+
+        if (toTest instanceof Subsystem && toTest.getSubsystemObject() != null) {
+            addRequirements(toTest.getSubsystemObject());
```

Comment:
You should be able to cast toTest to a subsystem instead of needing the get subsystem method. Django might not like this so check with him first. Would remove the call. Also if you do cast wrap it in a try catch.